services/QuillLMS/spec/models/teacher_activity_feed_spec.rb

```diff
@@ -0,0 +1,166 @@
+
+  describe '#text_for_score' do
+    let(:lesson_activity_session) { create(:lesson_activity_session) }
+    let(:diagnostic_activity_session) { create(:diagnostic_activity_session) }
+    let(:proficient_activity_session) { create(:connect_activity_session, percentage: 0.90) }
+    let(:nearly_proficient_activity_session) { create(:grammar_activity_session, percentage: 0.75) }
+    let(:not_yet_proficient_activity_session) { create(:proofreader_activity_session, percentage: 0.40) }
+
+    let(:feed) { TeacherActivityFeed.new('fake_id') }
+
+    context 'when the activity session is a lesson activity' do
+      it "returns #{ActivitySession::COMPLETED}" do
+        text = feed.send(:text_for_score, lesson_activity_session.classification.key, lesson_activity_session.percentage)
+
+        expect(text).to eq(ActivitySession::COMPLETED)
+      end
+    end
+
+    context 'when the activity session is a diagnostic activity' do
+      it "returns #{ActivitySession::COMPLETED}" do
+        text = feed.send(:text_for_score, diagnostic_activity_session.classification.key, diagnostic_activity_session.percentage)
+
+        expect(text).to eq(ActivitySession::COMPLETED)
+      end
+    end
+
+    context 'when the activity session percentage is at or above the proficiency cutoff' do
+      it "returns #{ActivitySession::PROFICIENT}" do
+        text = feed.send(:text_for_score, proficient_activity_session.classification.key, proficient_activity_session.percentage)
+
+        expect(text).to eq(ActivitySession::PROFICIENT)
+      end
+    end
+
+    context 'when the activity session percentage is at or above the nearly proficient cutoff' do
+      it "returns #{ActivitySession::NEARLY_PROFICIENT}" do
+        text = feed.send(:text_for_score, nearly_proficient_activity_session.classification.key, nearly_proficient_activity_session.percentage)
+
+        expect(text).to eq(ActivitySession::NEARLY_PROFICIENT)
+      end
+    end
+
+    context 'when the activity session percentage is below the nearly proficient cutoff' do
+      it "returns #{ActivitySession::NOT_YET_PROFICIENT}" do
+        text = feed.send(:text_for_score, not_yet_proficient_activity_session.classification.key, not_yet_proficient_activity_session.percentage)
+
+        expect(text).to eq(ActivitySession::NOT_YET_PROFICIENT)
+      end
+    end
+
+  end
+
+  describe '#text_for_completed' do
+    let(:feed) { TeacherActivityFeed.new('fake_id') }
+
+    context 'when the timestamp was less than one minute ago' do
+      it 'should return "0 mins ago"' do
+        expect(feed.send(:text_for_completed, 30.seconds.ago)).to eq("0 mins ago")
+      end
+    end
+
+    context 'when the timestamp was one minute ago' do
+      it 'should return "1 min ago"' do
+        expect(feed.send(:text_for_completed, 1.minute.ago)).to eq("1 min ago")
+      end
+    end
+
+    context 'when the timestamp was more than one minute but less than one hour ago' do
+      it 'should return "47 min ago"' do
```

Comment:
Not a real problem, but the "should" expectation here says "min" while the actual test is for "mins".

Reply:
Will change!